pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -354,14 +355,17 @@ func ReadLogs(ctx context.Context, path, containerID string, opts *LogOptions, r
 				if err != nil {
 					return err
 				}
-				if recreated {
+				if recreated || waitRecreate {
 					newF, err := os.Open(path)
 					if err != nil {
 						if os.IsNotExist(err) {
+							// If the new log file has not yet been created, try again to read
+							waitRecreate = true
 							continue
```

Comment:
Isn't it enough to just add a retry of `os.Open(path)` if os.IsNotExist(err)` happens?

I just want to confirm why we need to run through whole code(`r.ReadBytes()`, `f.Seek()` and `waitLogs()`) again with `continue`.
In case a log rotation happens, `r.ReadBytes()` and `f.Seek()` depend on an old file descripter and I feel it would be better to just retry `os.Open()` for a new file.

Reply:
@oomichi 
I thought that if a new file hasn't been created yet, there may be message still being written to the old file.
Therefore, I implemented the retry.
I have not confirmed whether such a situation actually occurs.

Reply:
@s-kawamura-w664 
Sorry for my late response, and thank you so much for explaining the senario.
That is reasonable to me.

/lgtm